File: backend/app/schemas/watchlist.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WatchlistCreate(BaseModel):
    symbol: str
    group_id: int | None = None
    note: str | None = None


class WatchlistRead(BaseModel):
    id: int
    tenant_id: str
    symbol: str
    group_id: int | None
    sort_order: int
    note: str | None
    is_pinned: bool
    is_special_attention: bool
    security_name: str
    security_code: str
    market: str
    tags: list[str]
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class WatchlistUpdate(BaseModel):
    group_id: int | None = None
    note: str | None = None
    is_pinned: bool | None = None
    is_special_attention: bool | None = None


class WatchlistReorderPayload(BaseModel):
    group_id: int
    pinned_ids: list[int]
    regular_ids: list[int]


class WatchlistGroupCreate(BaseModel):
    name: str


class WatchlistGroupUpdate(BaseModel):
    name: str | None = None


class WatchlistGroupReorderPayload(BaseModel):
    group_ids: list[int]


class WatchlistGroupRead(BaseModel):
    id: int
    tenant_id: str
    name: str
    is_system: bool
    sort_order: int
    item_count: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/watchlist.ts
export interface WatchlistItem {
  id: number
  tenant_id: string
  symbol: string
  group_id: number | null
  sort_order: number
  note: string | null
  is_pinned: boolean
  is_special_attention: boolean
  security_name: string
  security_code: string
  market: string
  tags: string[]
  created_at: string
}

export interface CreateWatchlistPayload {
  symbol: string
  group_id?: number | null
  note?: string | null
}

export interface UpdateWatchlistPayload {
  group_id?: number | null
  note?: string | null
  is_pinned?: boolean | null
  is_special_attention?: boolean | null
}

export interface ReorderWatchlistPayload {
  group_id: number
  pinned_ids: number[]
  regular_ids: number[]
}

export interface WatchlistGroup {
  id: number
  tenant_id: string
  name: string
  is_system: boolean
  sort_order: number
  item_count: number
  created_at: string
}

export interface CreateWatchlistGroupPayload {
  name: string
}

export interface UpdateWatchlistGroupPayload {
  name?: string
}

export interface ReorderWatchlistGroupsPayload {
  group_ids: number[]
}
